pennylane/ops/qubit/hamiltonian.py

```diff
@@ -550,3 +550,73 @@ def queue(self, context=qml.QueuingContext):
 
         context.append(self, owns=tuple(self.ops))
         return self
+
+    def sparse_matrix(self, wires=None):
+        """Sparse matrix representation of this Hamiltonian.
+
+        Args:
+            wires (Iterable): Wire labels that indicate the order of wires according to which the matrix
+             is constructed. If not provided, ``self.wires`` is used.
+
+        Returns:
+            .math.SparseMatrix: sparse matrix representation
+
+        **Example:**
+
+        If no wires are given, the matrix is constructed under the assumption that the wires are ordered
+        according to the Hamiltonian's wire ordering, which is inferred from the order of the operations.
+
+        >>> ham = qml.Hamiltonian(torch.tensor([0.1, 0.2]), [qml.PauliX('b'), qml.PauliY('a')])
+        >>> ham.wires
+        <Wires = ['b', 'a']>
+        >>> s = ham.sparse_matrix()
+        >>> print(s)
+        <SparseMatrix: entries=8, shape=(4, 4)>
+        >>> print(s.data)
+        {(0, 2): tensor(0.1), (1, 3): tensor(0.1), (2, 0): tensor(0.1), (3, 1): tensor(0.1),
+        (0, 1): tensor(0.-0.2j), (1, 0): tensor(0.+0.2j), (2, 3): tensor(0.-0.2j), (3, 2): tensor(0.+0.2j)}
+        >>> print(s.shape)
+        (4, 4)
+
+        Alternatively, one can provide a wire ordering. If no operation is found for a wire, an identity is inserted
+        at the position.
+
+        >>> ham = qml.Hamiltonian(torch.tensor([0.1]), [qml.PauliX('b')])
+        >>> s = ham.sparse_matrix(wires = ['a', 'b'])
+        >>> print(s.data)
+        {(0, 1): tensor(0.1), (1, 0): tensor(0.1), (2, 3): tensor(0.1), (3, 2): tensor(0.1)}
+        >>> print(s.shape)
+        (4, 4)
+        """
```

Comment:
This function does essentially the same as `utils.sparse_hamiltonian`: It computes the sparse representation of each constituent observable, and then linearly combines them.

Reply:
A lot of the logic can be outsourced to the Tensor class when we overhaul it!

Reply:
👍